packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -104,17 +114,139 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// This class stores the information of the selection under
+/// a [Selectable] or [SelectionHandler].
+///
+/// The [SelectedContentRange]s for a given [Selectable] or
+/// [SelectionHandler] can be retrieved by calling
+/// [SelectionHandler.getSelections].
+@immutable
+class SelectedContentRange {
+  /// Creates a [SelectedContentRange] with the given values.
+  const SelectedContentRange({
+    required this.contentLength,
+    required this.contentStart,
+    required this.startOffset,
+    required this.endOffset,
+  });
+
+  /// A selected content range that contains nothing.
+  static SelectedContentRange empty({int? contentLength}) {
+    return SelectedContentRange(
+      startOffset: -1,
+      endOffset: -1,
+      contentStart: -1,
+      contentLength: contentLength ?? -1,
+    );
+  }
+
+  /// The length of the content.
+  ///
+  /// The absolute value of the difference between the start
+  /// offset and end offset contained by this [SelectedContentRange]
+  /// must not exceed the content length.
+  final int contentLength;
+
+  /// The offset where the content begins.
+  final int contentStart;
+
+  /// The start of the selection relative to the start of the content.
+  ///
+  /// {@template flutter.rendering.selection.SelectedContentRange.selectionOffsets}
+  /// For example a [Text] widget's content is in the format
+  /// of an [InlineSpan] tree.
+  ///
+  /// Take the [Text] widget and [TextSpan] tree below:
+  ///
+  /// {@tool snippet}
+  /// ```dart
+  /// const Text.rich(
+  ///   TextSpan(
+  ///     text: 'Hello world, ',
+  ///     children: <InlineSpan>[
+  ///       WidgetSpan(
+  ///         child: Text('how are you today?'),
+  ///       ),
+  ///     ],
+  ///   ),
+  /// )
+  /// ```
+  /// {@end-tool}
+  ///
+  /// If we select from the beginning of 'world' to the
+  /// end of the '?' in the [WidgetSpan], we will receive
+  /// two [SelectedContentRange]s from [SelectionHandler.getSelections].
+  /// The first range will be relative to the root text
+  /// of the [TextSpan], the [startOffset] will be 6,
+  /// and [endOffset] will be 13. The second range
+  /// will be relative to the content inside the
+  /// [WidgetSpan], the [startOffset] will be 0,
+  /// and [endOffset] will be 18.
+  /// {@endtemplate}
+  final int startOffset;
+
+  /// The end of the selection relative to the end of the content.
+  ///
+  /// {@macro flutter.rendering.selection.SelectedContentRange.selectionOffsets}
+  final int endOffset;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
```

Comment:
Make sure you didn't miss this :arrow_up:

Reply:
I wonder if this can be left not final. That way someone could extend this class and add supplementary identifiers, if they wanted to go that route. `SelectionDetails` has been made final.